test/e2e/tests/ratelimit.go

```diff
@@ -662,6 +669,68 @@ var RateLimitHeadersAndCIDRMatchTest = suite.ConformanceTest{
 	},
 }
 
+var UsageRateLimitTest = suite.ConformanceTest{
```

Comment:
this is almost identical with the usecase with ai-gateway - except that the extproc sets the dynamic metadata on responseHeaders hook vs ai-gateway in responseBody hook. Either should work from Envoy pov as the response cost is applied when stream is closing